### PR TITLE
Update video player rail and track

### PR DESCRIPTION
**Pull request by sudhons**
The PR is a reimplementation/refactoring of the video player's Track and Rail. The previously (now unused) code are deleted.

The new implementation doesn't calculate overlapping segments; it overlaps the highlights scrub bar instead.

### CODE REVIEW

File: src/components/progress-bar/components/Rail.tsx [hunk 1]
```diff
@@ -1,28 +1,32 @@
...
-			getHighlightColorBlended={getHighlightColorBlended}
-		/>
+		<div className={sliderRail}>
+			{highlights?.map(({ id, colors, start, end }) => (
+				<RailStyled
+					key={id}
+					startPoint={getPercentFromDuration(start, videoDuration)}
+					width={getPercentFromDuration(end - start, videoDuration)}
+					color={getHighlightColorBlended?.(colors)}
+					startColorSegment={getHighlightColorBlended?.(colors, BLEND_CONFIG)}
```
Comment (ronnyroeller) [inline src/components/progress-bar/components/Rail.tsx:26]: Extract this into a variable and use for `startColorSegment` and `endColorSegment`. This communicates clearer that this is supposed to be the same value.

**Review by ronnyroeller — APPROVED** [review]
LGTM (pls see comment)
Pls also add a comment to the PR what it does: there seems to be a lot of code deleted - to where was this moved?

**Comment by sudhons** [comment]
@koyewonqw These are screenshots of how it looks before and after the refactoring:
The colors used are:
- highlight 1: blending of #08E8E8 and #F4DF82
- highlight 2: blending of #89CFF0 and #FC6399
- highlight 3: just #FF6347

## Results
The result shows three cases: 

### Case One: Individual highlights
Before                                  |  After
:-------------------------:|:-------------------------:
Screenshot 2022-09-13 at 14 13 44 | Screenshot 2022-09-13 at 13 53 43
Screenshot 2022-09-13 at 14 14 35 | Screenshot 2022-09-13 at 13 53 58
Screenshot 2022-09-13 at 14 14 45 | Screenshot 2022-09-13 at 13 54 14
Screenshot 2022-09-13 at 14 14 55 | Screenshot 2022-09-13 at 13 54 33


### Case Two: Overlapping highlights
Before                                  |  After
:-------------------------:|:-------------------------:
Screenshot 2022-09-13 at 14 07 36 | Screenshot 2022-09-13 at 13 59 11
Screenshot 2022-09-13 at 14 07 51 | Screenshot 2022-09-13 at 13 59 22
Screenshot 2022-09-13 at 14 08 09 | Screenshot 2022-09-13 at 13 59 39


### Case Three: Continuous highlights(i.e one highlights starts where the previous ends)
Before                                  |  After
:-------------------------:|:-------------------------:
Screenshot 2022-09-13 at 14 04 50        |  Screenshot 2022-09-13 at 14 01 55

**Review by koyewonqw — APPROVED** [review]
Look good to me thanks

**Pull request merged**